src/google/protobuf/arena.cc

```diff
@@ -453,7 +453,7 @@ ThreadSafeArena::ThreadCache& ThreadSafeArena::thread_cache() {
   return thread_cache;
 }
 #else
-ABSL_CONST_INIT PROTOBUF_THREAD_LOCAL
```

Comment:
Can we change this to PROTOBUF_CONSTINIT?  That handles this MSVC issue we've encountered before, without affecting linux

Reply:
e.g. #8688 and #9698 are two other cases we hit this

Reply:
Yes, changing ABSL_CONST_INIT -> PROTOBUF_CONSTINIT fixed the issue! I've updated the PR